File: benchmark-ECD/benchmark-QuantumOptics/bmutils.jl
module bmutils
using QuantumOptics
using JSON

# numerical thingy to set all subnormal numbers to zero
set_zero_subnormals(true)

function save(name, results)
    result_path = "C:/Users/jonat/desktop/Code/multimode/benchmark-ECD/results/julia-$name.json"
    f = open(result_path, "w")
    write(f, JSON.json(results))
    close(f)
end

end # module

File: benchmark-ECD/benchmark-QuTiP/bm_utils.py
import timeit
import json


result_path = "C:/Users/jonat/Desktop/Code/multimode/benchmark-ECD/results/results-QuTiP-{}.json"


def run_benchmark(f, *args, samples=5, evals=1):
    D = {"f": f, "args": args}
    t = timeit.repeat("f(*args)", globals=D, number=evals, repeat=samples)
    return min(t)/evals


def save(name, results):
    f = open(result_path.format(name), "w")
    json.dump(results, f)
    f.close()